examples/swash_render/src/main.rs

```diff
@@ -0,0 +1,223 @@
+    let output_path = {
+        let path = std::path::PathBuf::from(file!());
+        let mut path = std::fs::canonicalize(path).unwrap();
+        path.pop();
+        path.pop();
+        path.pop();
+        path.push("_output");
+        let _ = std::fs::create_dir(path.clone());
+        path.push("swash_render.png");
+        path
+    };
+    let output_file = File::create(output_path).unwrap();
+    let png_encoder = PngEncoder::new(output_file);
+    img.write_with_encoder(png_encoder).unwrap();
+}
+
+fn render_glyph_run(
+    context: &mut ScaleContext,
+    glyph_run: &GlyphRun<Color>,
+    img: &mut RgbaImage,
+    padding: u32,
+) {
+    // Resolve properties of the GlyphRun
+    let mut run_x = glyph_run.offset();
+    let run_y = glyph_run.baseline();
+    let style = glyph_run.style();
+    let color = style.brush;
+
+    // Get the "Run" from the "GlyphRun"
+    let run = glyph_run.run();
+
+    // Resolve properties of the Run
+    let font = run.font();
+    let font_size = run.font_size();
+    let normalized_coords = run.normalized_coords();
+
+    // Convert from parley::Font to swash::FontRef
+    let font_ref = FontRef::from_index(font.data.as_ref(), font.index as usize).unwrap();
+
+    // Iterates over the glyphs in the GlyphRun
+    for glyph in glyph_run.glyphs() {
+        let glyph_id: GlyphId = glyph.id;
+        let glyph_x = run_x + glyph.x;
+        let glyph_y = run_y - glyph.y;
+        run_x += glyph.advance;
+        let Some(rendered_glyph) = render_glyph(
+            context,
+            &font_ref,
+            font_size,
+            normalized_coords,
+            glyph_id,
+            glyph_x.fract(),
+            glyph_y.fract(),
+        ) else {
+            println!("No glyph");
+            continue;
+        };
+
+        let glyph_width = usize::try_from(rendered_glyph.placement.width).expect("usize < 32 bits");
+        let glyph_height =
+            usize::try_from(rendered_glyph.placement.height).expect("usize < 32 bits");
+        let glyph_origin_x =
+            glyph_x.floor() as i32 + rendered_glyph.placement.left + padding as i32;
+        let glyph_origin_y =
+            (glyph_y.floor() as i32) - rendered_glyph.placement.top + padding as i32;
+
+        match rendered_glyph.content {
+            Content::Mask => {
+                let mut i = 0;
+                for off_y in 0..glyph_height as i32 {
+                    for off_x in 0..glyph_width as i32 {
+                        let x = (glyph_origin_x + off_x) as u32;
+                        let y = (glyph_origin_y + off_y) as u32;
+                        let alpha = rendered_glyph.data[i];
+                        let color = Rgba([color.r, color.g, color.b, alpha]);
+                        img.get_pixel_mut(x, y).blend(&color);
+                        i += 1;
+                    }
+                }
+            }
+            Content::SubpixelMask => unimplemented!(),
```

Comment:
@dfrg What are the semantics of the `SubpixelMask`? (/ how would one apply that to an RGBA image?). I reckon it would be good to demonstrate that even if we don't actually enable subpixel mode in the example.

Reply:
This is an rgba image with an alpha component per color channel (the alpha in the mask is ignored). This document (https://github.com/servo/webrender/blob/main/webrender/doc/text-rendering.md#subpixel-text-blending) has a good explanation of the math.